### PR TITLE
Throwables.hasStackTraceContaining(String) 

### PR DESCRIPTION
**Pull request by DanielZlotin**
see issue #607 
Also added util.Throwables.getStackTrace(Throwable) which converts the stack trace to String

### CODE REVIEW

File: src/main/java/org/assertj/core/util/Throwables.java [hunk 1]
```diff
@@ -114,6 +116,26 @@ public static Throwable getRootCause(Throwable throwable) {
...
+   * @param throwable the {@code Throwable} to get stack trace from.
+   * @return the stack trace as a {@link String}.
+   * 
+   * @author Daniel Zlotin
+   */
+  public static String getStackTrace(Throwable throwable) {
+    final StringWriter sw = new StringWriter();
+    final PrintWriter pw = new PrintWriter(sw, true);
+    throwable.printStackTrace(pw);
+    return sw.getBuffer().toString();
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/util/Throwables.java]: I'm not totally sure it is necessary but I would close the writer or use a  try-with-resources.

File: src/test/java/org/assertj/core/internal/throwables/Throwables_assertHasStackTraceContaining_Test.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  @Test
+  public void should_fail_if_actual_is_null() {
+    thrown.expectAssertionError(actualIsNull());
+    throwables.assertHasStackTraceContaining(someInfo(), null, "Throwable");
+  }
+
+  @Test
+  public void should_pass_if_actual_has_stacktrace_containing_with_expected_description() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/throwables/Throwables_assertHasStackTraceContaining_Test.java]: `containing_with_expected` ->  `containing_the_expected`

File: src/main/java/org/assertj/core/util/Throwables.java (hunk 1 above)
Comment (DanielZlotin) [inline src/main/java/org/assertj/core/util/Throwables.java]: Closing a StringWriter does nothing, and closing PrintWriter just delegates to StringWriter::close

**Comment by DanielZlotin** [comment]
@joel-costigliola done

**Comment by joel-costigliola** [comment]
Will integrate it after my holidays :-)

**Comment by joel-costigliola** [comment]
Integrated thanks. (will be in 2.4.0 and 3.4.0)

**Pull request closed without merging**